airbyte-integrations/connectors/destination-mysql/src/main/java/io/airbyte/integrations/destination/mysql/MySQLDestination.java

```diff
@@ -112,6 +122,55 @@ public JsonNode toJdbcConfig(final JsonNode config) {
     return Jsons.jsonNode(configBuilder.build());
   }
 
+  private List<String> getAdditionalParameters(final JsonNode config) {
+    final Map<String, String> customParameters = getCustomJdbcParameters(config);
+
+    if (useSSL(config)) {
+      return convertToJdbcStrings(customParameters, List.of(DEFAULT_JDBC_PARAMETERS, SSL_JDBC_PARAMETERS));
+    } else {
+      return convertToJdbcStrings(customParameters, List.of(DEFAULT_JDBC_PARAMETERS));
+    }
+  }
+
+  private List<String> convertToJdbcStrings(final Map<String, String> customParameters, final List<Map<String, String>> maps) {
+    final Set<String> keys = maps.stream()
+        .map(Map::keySet)
+        .flatMap(Collection::stream)
+        .collect(Collectors.toSet());
+    final List<String> duplicateKeys = keys.stream().filter(customParameters::containsKey).toList();
+    if (!duplicateKeys.isEmpty()) {
+      throw new RuntimeException("Cannot overwrite default JDBC parameter " + duplicateKeys);
+    }
+    return Streams.concat(Stream.of(customParameters), maps.stream())
+        .map(Map::entrySet)
+        .flatMap(Collection::stream)
+        .map(entry -> formatParameter(entry.getKey(), entry.getValue()))
+        .collect(Collectors.toList());
+  }
+
+  private Map<String, String> getCustomJdbcParameters(final JsonNode config) {
+    final Map<String, String> parameters = new HashMap<>();
+    if (config.has(JDBC_URL_PARAMS_KEY)) {
+      final String jdbcParams = config.get(JDBC_URL_PARAMS_KEY).asText();
+      if (!jdbcParams.isBlank()) {
+        final String[] keyValuePairs = jdbcParams.split("&");
```

Comment:
@tuliren do you think it's too much work the parse the string into a key value map? It makes deduplicating the keys with the default parameters more straightforward.

Reply:
I like this solution. Agree that this makes the assertion method simpler.